src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -665,7 +665,7 @@ trait Namers extends MethodSynthesis {
       else assignAndEnterFinishedSymbol(tree)
 
       if (isEnumConstant(tree)) {
-        tree.symbol setInfo ConstantType(Constant(tree.symbol))
+        //tree.symbol setInfo ConstantType(Constant(tree.symbol))
```

Comment:
Currently some tests fail with `annotation argument needs to be a constant; found: Enum.VALUE`, but enabling this causes:

```
Week.scala:20: error: type mismatch;
 found   : foo.Week.type(Saturday)
 required: foo.Week
```